### PR TITLE
GetActiveOrders/GetOrderHistory wrapper implementation

### PR DESCRIPTION
**Pull request by gloriousCode**
# Description

Adds GetActiveOrders + GetOrderHistory API wrapper support, specifically 
`GetActiveOrders(getOrdersRequest exchange.GetOrdersRequest) ([]exchange.OrderDetail, error)` and
`GetOrderHistory(getOrdersRequest exchange.GetOrdersRequest) ([]exchange.OrderDetail, error)`. 
## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Two tests have been added to all exchange test files `TestGetActiveOrders(t *testing.T)` and `TestGetOrderHistory(t *testing.T)`. All pass when no keys are present. Below is a detailed breakdown of supported endpoints, testing and notes:

Exchange |  Implemented | Tested | Notes
------------ | --- | ---  | ------------- 
Alphapoint | y | n | No account to test properly
ANX | y | n | UNAUTHORISED. I believe my account has been flagged for spamming
Binance | y | y | 
Bitfinex | y | y | Undocumented issue regarding inactive order history saying an incorrect pair is submitted
Bitflyer | n | y | Bitflyer authenticated endpoints not implemented. No ability to create account
Bithumb | y | y | 
Bitmex | y | y | 
Bitstamp | y | y | 
Bittrex | y | y | 
BTCC | n | y | 
BTCMarkets | y | y | 
CoinbasePro | y | n | Cannot use API without full verification
Coinut | y | y | 
Exmo | y | y | 
GateIO | y | y | 
Gemini | y | y |
HitBTC | y | y | 
Huobi | y | y | 
HuobiHadax | y | y | 
ItBit | y | n | ItBit have been processing my verification for awhile
Kraken | y | y | 
LakeBTC | y | y | Possibility that orderid parameter is required to retrieve data. However,  response is fine (I have no orders to verify)
Liqui | y | n | Cannot properly sign up to Liqui
Localbitcoins | y | n | Cannot use API without full verification
OKCoin | y | y | 
OKEx | y | y | 
Poloniex | y | y | 
Wex | y | n | _Wexecuted_ 🔫 
Yobit | y | y | 
ZB | y | y | 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool 
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Travis with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/alphapoint/alphapoint_wrapper.go [hunk 1]
```diff
@@ -216,3 +216,117 @@ func (a *Alphapoint) GetFeeByType(feeBuilder exchange.FeeBuilder) (float64, erro
...
+// GetActiveOrders retrieves any orders that are active/open
+func (a *Alphapoint) GetActiveOrders(getOrdersRequest exchange.GetOrdersRequest) ([]exchange.OrderDetail, error) {
+	resp, err := a.GetOrders()
+	if err != nil {
+		return nil, err
+	}
+
+	var orders []exchange.OrderDetail
+	for x := range resp {
+		for _, order := range resp[x].Openorders {
```
Comment (thrasher-) [inline exchanges/alphapoint/alphapoint_wrapper.go]: Please change to OpenOrders

File: currency/pair/pair.go [hunk 2]
```diff
@@ -108,6 +108,15 @@ func NewCurrencyPair(firstCurrency, secondCurrency string) CurrencyPair {
 	}
 }
 
+// NewCurrencyPairWithDelimiter returns a CurrencyPair with a delimiter
+func NewCurrencyPairWithDelimiter(firstCurrency, secondCurrency, delimiter string) CurrencyPair {
```
[the remark this answers is not in the document]
Comment (thrasher-) [reply currency/pair/pair.go]: Oh right, that's fine to leave as is then

File: exchanges/wex/wex_types.go [hunk 3]
```diff
@@ -51,6 +51,7 @@ type ActiveOrders struct {
 	Rate             float64 `json:"rate"`
 	TimestampCreated float64 `json:"time_created"`
 	Status           int     `json:"status"`
+	ID               string
```
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/wex/wex_types.go]: For all of these instances, the ID is returned as the key in a map and not in the JSON object. I've added this to allow the data to be accessible after appending all the order data together for processing in the wrapper (see [here](https://github.com/thrasher-/gocryptotrader/pull/239/files#diff-8fe624a4427c538e17b8b9b39087cc33R289) I'm always open to other ideas for passing data, but I think this is a simple approach

File: exchanges/alphapoint/alphapoint_wrapper.go [hunk 4]
```diff
@@ -216,3 +216,117 @@ func (a *Alphapoint) GetFeeByType(feeBuilder exchange.FeeBuilder) (float64, erro
...
+				orderDetail.OrderSide = string(exchange.SellOrderSide)
+			}
+
+			switch order.OrderType {
+			case 1:
+				orderDetail.OrderType = string(exchange.MarketOrderType)
+			case 2:
+				orderDetail.OrderType = string(exchange.LimitOrderType)
+			case 3:
+				fallthrough
```
Comment (xtda) [inline exchanges/alphapoint/alphapoint_wrapper.go]: Is there ever going to be a case where this is used?
Go supports multiple comma separated values on a switch 
Is it worth grouping 3,4 & 5,6? 

Might even be worth moving this over to a map there is cases where a map lookup on small data set is faster than a switch obviously some benchmarking might be needed to see if its even worth it and not premature optimization

Could also be used to remove some code duplication
https://github.com/thrasher-/gocryptotrader/blob/obtain-older-orders/exchanges/exchange.go#L826
Could be replaced with a map like

```
orderType := map[int]string{
		0: "ANY",
		1: "LIMIT",
		2: "MARKET",
		3: "IMMEDIATE_OR_CANCEL",
		4: "STOP",
		5: "TRAILINGSTOP",
		6: "UNKNOWN",
	}
```
and remove the additional switch and just do a lookup of the map
```
orderDetail.OrderType = orderType[order.OrderType]
```

Would most likely want to add some error checking to it to make sure the value is found/valid
A map lookup will return value, value found (boolean)

File: exchanges/wex/wex_types.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (shazbert) [reply exchanges/wex/wex_types.go]: OKay fair enough, I usually do things different I would have resp struct then create a struct 2 for return. No need to clarify more.

File: exchanges/hitbtc/hitbtc_types.go [hunk 5]
```diff
@@ -184,6 +184,23 @@ type AuthenticatedTradeHistoryResponse struct {
...
+	ClientOrderID string  `json:"clientOrderId"`
+	Symbol        string  `json:"symbol"`
+	Side          string  `json:"side"`
+	Status        string  `json:"status"`
+	Type          string  `json:"type"`
+	TimeInForce   string  `json:"timeInForce"`
+	Price         float64 `json:"price,string"`
+	Quantity      float64 `json:"quantity,string"`
+	PostOnly      bool    `json:"postOnly"`
+	CumQuantity   float64 `json:"cumQuantity,string"`
```
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/hitbtc/hitbtc_types.go]: What do you mean by this?

File: exchanges/hitbtc/hitbtc_wrapper.go [hunk 6]
```diff
@@ -264,3 +266,104 @@ func (h *HitBTC) GetFeeByType(feeBuilder exchange.FeeBuilder) (float64, error) {
...
+		for _, order := range resp {
+			allOrders = append(allOrders, order)
+		}
+	}
+
+	var orders []exchange.OrderDetail
+	for _, order := range allOrders {
+		t, err := time.Parse(time.RFC3339, order.CreatedAt)
+		if err != nil {
+			log.Errorf("Exchange %v Func %v Order %v Could not parse date to unix with value of %v",
```
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/hitbtc/hitbtc_wrapper.go]: >  or can we cahnge all parses to errors?
What do you mean by this?

[the remark this answers is not in the document]
Comment (shazbert) [reply exchanges/hitbtc/hitbtc_wrapper.go]: Some logs are Warnf and some are Errorf for this issue. Just make them all the same.

File: exchanges/alphapoint/alphapoint_wrapper.go (hunk 4 above)
Comment (gloriousCode) [reply exchanges/alphapoint/alphapoint_wrapper.go]: I didn't know about the switch commas, thanks!
Order types are already used when submitting an order. While not all exchanges return the type of order in this PR, 12 do, so its worth keeping the data if its returned.

I'll look at using a map for it all now

File: exchanges/yobit/yobit_test.go [hunk 7]
```diff
@@ -35,6 +37,7 @@ func TestSetup(t *testing.T) {
 	conf.AuthenticatedAPISupport = true
 
 	y.Setup(conf)
+
```
Comment (shazbert) [inline exchanges/yobit/yobit_test.go]: RM line

**Comment by shazbert** [comment]
utACK

File: exchanges/exchange.go [hunk 8]
```diff
@@ -945,3 +958,110 @@ func (e *Base) FormatWithdrawPermissions() string {
...
+	CancelledOrderStatus       OrderStatus = "CANCELED"
+	PendingCancelOrderStatus   OrderStatus = "PENDING_CANCEL"
+	RejectedOrderStatus        OrderStatus = "REJECTED"
+	ExpiredOrderStatus         OrderStatus = "EXPIRED"
+	HiddenOrderStatus          OrderStatus = "HIDDEN"
+	UnknownOrderStatus         OrderStatus = "UNKNOWN"
+)
+
+// FilterOrdersBySide removes any OrderDetails that don't match the orderStatus provided
+func (e *Base) FilterOrdersBySide(orders *[]OrderDetail, orderSide OrderSide) {
```
Comment (thrasher-) [inline exchanges/exchange.go]: Better performance can be achieved by using go's built in sort package. Please see this as a reference: 

https://github.com/thrasher-/gocryptotrader/blob/master/exchanges/stats/stats.go#L24

File: exchanges/bitmex/bitmex_wrapper.go [hunk 9]
```diff
@@ -304,3 +304,82 @@ func (b *Bitmex) GetFeeByType(feeBuilder exchange.FeeBuilder) (float64, error) {
...
+	params := OrdersRequest{}
+	params.Filter = "{\"open\":true}"
+
+	resp, err := b.GetOrders(params)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, order := range resp {
+		orderSide := orderSideMap[order.Side]
```
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/bitmex/bitmex_wrapper.go]: Where do you think would be the best place to put it?
On the declaration of orderSideMap, above the use of `orderSide` in the func, or the declaration of the func that uses the map?

[the remark this answers is not in the document]
Comment (thrasher-) [reply exchanges/bitmex/bitmex_wrapper.go]: As a comment in all the functions declarations which use the map

**Pull request merged**